app/furniture/marketplace/delivery/window.rb

```diff
@@ -0,0 +1,29 @@
+class Marketplace
+  class Delivery
+    class Window < Model
+      attr_accessor :value
+      delegate(:nil?, :blank?, :empty?, to: :value)
+
+      def strftime(*args, **kwargs)
+        return value.strftime(*args, **kwargs) if time_like?
+        value
+      end
+
+      alias_method :to_s, :strftime
+
+      def eql?(other)
+        if other.is_a?(Window)
+          value.eql?(other.value)
+        else
+          eql?(WindowType.new.cast(other))
+        end
+      end
+
+      alias_method :==, :eql?
+
+      def time_like?
+        value.respond_to?(:strftime)
```

Comment:
Nit: maybe inlining this method would make the `strftime` method clearer -- you are calling strftime if it responds to it -- saving the extra indirection of going "is `time_like?` a built-in thing? oh, no, it is defined down here, let's see... oh, i see".

Reply:
Yea, there's an indirection bump but `time_like?` is used in a few other places (in particular figuring out which field to render for the form and how to present the window on the receipt)

Reply:
Ah, fair enough.